content/400-reference/200-api-reference/050-prisma-client-reference.mdx

```diff
@@ -2004,25 +2004,40 @@ Sorts a list of records. See also: [Sorting](/concepts/components/prisma-client/
 
 #### Remarks
 
-- In [2.16.0](https://github.com/prisma/prisma/releases/2.16.0) and later, you can [order by relation fields](#sort-post-by-the-related-user-records-name) (Preview) - for example, order posts by the author's name.
+- In [2.16.0](https://github.com/prisma/prisma/releases/2.16.0) and later, you can [order by relation fields](#sort-post-by-the-related-user-records-name) - for example, order posts by the author's name.
 
-- In [3.5.0](https://github.com/prisma/prisma/releases/3.5.0) and later, in PostgreSQL you can [order by relevance](#sort-post-by-relevance-of-the-title). For more details, see [Sort by relevance](/concepts/components/prisma-client/filtering-and-sorting#sort-by-relevance-postgresql).
+- In [3.5.0](https://github.com/prisma/prisma/releases/3.5.0) and later, in PostgreSQL you can [order by relevance](#sort-post-by-relevance-of-the-title). For details, see [Sort by relevance](/concepts/components/prisma-client/filtering-and-sorting#sort-by-relevance-postgresql).
 
-#### Inputs
+- In [4.1.0](https://github.com/prisma/prisma/releases/4.1.0) and later, you can [sort `null` records first or last](#sort-post-by-the-related-user-records-name-with-null-records-first). For details, see [Sort with nulls first or last](/concepts/components/prisma-client/filtering-and-sorting#sort-with-null-records-first-or-last).
+
+#### Inputs for `sort` argument
 
 | Name   | Description                  |
 | ------ | ---------------------------- |
 | `asc`  | Sort ascending (A &rarr; Z)  |
 | `desc` | Sort descending (Z &rarr; A) |
 
+#### Inputs for `nulls` argument
+
+Note:
+
+- This argument is optional.
+- It is for use on optional fields only. If you try to sort by nulls on a mandatory field, Prisma Client throws a [P2009 error](/reference/api-reference/error-reference#p2009).
```

Comment:
Nit pick honestly, but being specific might be useful 👇

```suggestion
- It is for use on optional scalar fields only. If you try to sort by nulls on a mandatory field, Prisma Client throws a [P2009 error](/reference/api-reference/error-reference#p2009).
```

One more thing, is `mandatory` the word we use throughout the docs to refer to non-optional fields? This is an open question as we usually refer to them as `required` fields internally

Reply:
Good point! This is one for our terminology list

Reply:
Thanks. I'll rework this paragraph to mention relation fields and required, rather than mandatory.

Reply:
I've brought in these suggestions, so I'm closing the comment